shared/constants/types/chat2/index.js

```diff
@@ -30,7 +35,7 @@ export type _State = {
   messageOrdinals: I.Map<Common.ConversationIDKey, I.SortedSet<Message.Ordinal>>, // ordered ordinals in a thread
   metaMap: I.Map<Common.ConversationIDKey, Meta.ConversationMeta>, // metadata about a thread
   explodingModes: I.Map<Common.ConversationIDKey, number>, // seconds to exploding message expiration
-  quotingMap: I.Map<string, _quotedOrdConv>, // current message being quoted
+  quote: ?QuoteInfo, // last quoted message
```

Comment:
Quoting is an action, so we only need to keep around the last one. Editing is both a state (editing or not editing) and an action (inject / clear text to edit) so it still needs to be a map for the state part.